### PR TITLE
 Add the MapTiler provider to Terramap + minor translations es_es

### PR DESCRIPTION
**Pull request by hugofnm**
I'm not sure for display_priority...

### CODE REVIEW

**Comment by SmylerMC** [comment]
Display priority is used to determine the order in which to show the styles in the style menu. It's also used to choose the fallback style for the minimap when the style specified in the config is not found.

Aside from that, there are a few issues with your pull request:
  - There is already a satellite style available, adding one is not really necessary, even if it's hybrid. Mapbox's free tier allows for more request per month (750k against 100k).
  - I can't really add that style without being available to manage the api key myself, and it's not really convenient for me to be checking two different key from two different services regularly. 
  - Adding the style as a place-holder for people to add their own keys is not an option. This is the master style file, it's included in the jar file as a fallback and downloaded by the mod when the game starts. All  entries here are required to be valid, working maps. Personal changes should be done  in the user config file.
  - Allowing such a style on the minimap is problematic: it first means more traffic, of course, but there is an other problem with it, which is that for obvious screen pace reasons the copyright string is not rendered on the minimap. OSM has some flexibility when it comes to this kind of problems when they get credited elsewhere, so I took the liberty to have osm maps displayed without their copyrights on the minimap. I don't think that liberty can be taken with professional services such as Mapbox and MapTiler, which is why the mapbox style has `allow_on_minimap` set to false, and so should it be for MapTiler.
  - Any changes to this file should increase the version number in the metadata section.

For those reasons, I cannot accept your changes to the master map file regarding the MapTiler style, but you are of course free to define the styles you want in your `config/terramap_user_styles.json` file for personal use. You are responsible for what you put there.

Your language related changes are of course welcome.

So in conclusion, please remove the MapTiler related changes and increment the version number. I'll keep a copy of your MapTiler template in case I ever choose to switch from Mapbox to MapTiler, but I can't include them for the moment, sorry.
